test/Kestrel.InMemory.FunctionalTests/RequestTests.cs

```diff
@@ -67,10 +67,9 @@ public async Task RequestBodyReadAsyncCanBeCancelled()
                 var buffer = new byte[1024];
                 try
                 {
+                    await context.Request.Body.ReadUntilLengthAsync(buffer, 6, cts.Token).DefaultTimeout();
 
-                    int read = await context.Request.Body.ReadAsync(buffer, 0, buffer.Length, cts.Token);
-
-                    Assert.Equal("Hello ", Encoding.UTF8.GetString(buffer, 0, read));
+                    Assert.Equal("Hello ", Encoding.ASCII.GetString(buffer, 0, 6));
```

Comment:
Only this change and the change to TraceIdentifierIsUnique() fix any real test bugs. The rest is minor test simplification.

The only other place I see where we read the request body without a loop is in Http2Stream/ConnectionTests. There's a lot of that there, but I think that's OK as long as we don't start adding data to the request body pipe for partial frames.